vm/devices/storage/disk_nvme/nvme_driver/src/tests.rs

```diff
@@ -126,3 +134,70 @@ async fn test_nvme_driver(driver: DefaultDriver) {
 
     driver.shutdown().await;
 }
+
+#[async_test]
+async fn test_nvme_save_restore(driver: DefaultDriver) {
```

Comment:
Thanks for adding tests here, Yuri. Would you also add some test cases for:

- Multiple namespaces per controller
- Failure to restore (at various stages)
- A save/restore with pending IO

I don't think we should block this PR on these changes, as long as there's a plan to follow up. I also understand some of these tests may be difficult to write with the current architecture. We can also use this discussion to identify those gaps.

Reply:
Created https://github.com/microsoft/openvmm/issues/429 to track this effort